### PR TITLE
fix: Add warning when x-example is not of correct type

### PR DESCRIPTION
**Pull request by pksunkara**
If you look at the validation for defaultValue, we do not care about number and boolean validations. So, I haven't done it for sample value too. I have only fixed the array validation for now because there are other validations we want to do when we want to start on it. It's better as a separate task.

### CODE REVIEW

File: src/parser.js [hunk 1]
```diff
@@ -1157,12 +1158,19 @@ export default class Parser {
       Type = BooleanElement;
     } else if (parameter.type === 'array') {
       Type = ArrayElement;
+
+      if (example && !Array.isArray(example)) {
```
Comment (kylef) [inline src/parser.js]: The check is only for array types, I think we could make this generic and support all types. What do you think?

Mostly (possibly working) pseudo code but you get the idea:

```js
const parameterTypes = {
  string: {
    element: StringElement,
    type: String
  },
  array: {
    element: ArrayElement,
    type: Array
  },
  // all types go here
}

const parameterType = parameterTypes[parameter.type];
if (parameterType) {
  const element = new parameterType.element();

  if (example) {
    if (example instance parameterType.type) {
      element.content = example;
      // todo set source map?
    } else {
      // create annotation of example being wrong type
    }
  }
} else {
  // create annotation of unsupported/unknown type
}
```

**Comment by kylef** [comment]
@pksunkara Can you add a CHANGELOG entry please.

**Comment by pksunkara** [comment]
That's what I wrote in the description. It's not just limited to sample value. We need to do the checks elsewhere too. Which is why it's better as a separate task.

**Comment by pksunkara** [comment]
Re: changelog, I wanted to leave it to Thomas as part of his release task. And then introduce him about making it part of PRs.

**Pull request merged**